lib/main_common.dart

```diff
@@ -1,46 +1,61 @@
     AppConfig config = AppConfig.of(context);
-    return _buildApp(config.baseUrl);
+    return _buildApp(context, config.baseUrl);
   }
 
-  Widget _buildApp(String baseUrl) {
-    return MaterialApp(
-        title: 'Carriage',
-        theme: ThemeData(
-            primarySwatch: Colors.red,
-            fontFamily: 'SFPro',
-            accentColor: Color.fromRGBO(60, 60, 67, 0.6),
-            textTheme: TextTheme(
-                headline: TextStyle(fontSize: 36, fontWeight: FontWeight.bold),
-                subhead: TextStyle(fontSize: 17.0, fontWeight: FontWeight.bold),
-                display1:
-                    TextStyle(fontSize: 12.0, fontWeight: FontWeight.bold),
-                display2:
-                    TextStyle(fontSize: 12.0, fontWeight: FontWeight.normal),
-                display3:
-                    TextStyle(fontSize: 16.0, fontWeight: FontWeight.normal))),
-        home: Login());
+  Widget _buildApp(BuildContext context, String baseUrl) {
+    return MultiProvider(
+      providers: [
+        ChangeNotifierProvider<AuthProvider>(create: (BuildContext context) {
+          return AuthProvider(context);
+        })
+      ],
+      child: MaterialApp(
+          title: 'Carriage',
+          theme: ThemeData(
+              primarySwatch: Colors.red,
+              fontFamily: 'SFPro',
+              accentColor: Color.fromRGBO(60, 60, 67, 0.6),
+              textTheme: TextTheme(
+                  headline:
+                      TextStyle(fontSize: 36, fontWeight: FontWeight.bold),
+                  subhead:
+                      TextStyle(fontSize: 17.0, fontWeight: FontWeight.bold),
+                  display1:
+                      TextStyle(fontSize: 12.0, fontWeight: FontWeight.bold),
+                  display2:
+                      TextStyle(fontSize: 12.0, fontWeight: FontWeight.normal),
+                  display3: TextStyle(
+                      fontSize: 16.0, fontWeight: FontWeight.normal))),
+          home: Logic()),
+    );
   }
 }
+class Logic extends StatelessWidget {
+  Logic({Key key}) : super(key: key);
 
-authenticationRequest(String baseUrl, String token, String email) async {
-  String endpoint = baseUrl + '/auth';
-  var requestBody = {
-    "token": token,
-    "email": email,
-    "clientID":
-        "241748771473-0r3v31qcthi2kj09e5qk96mhsm5omrvr.apps.googleusercontent.com",
-    "table": "Drivers"
-  };
-  Response response = await post(endpoint, body: requestBody);
-  return response.body;
-}
+  @override
+  Widget build(BuildContext context) {
+      AuthProvider p = Provider.of<AuthProvider>(context);
+      if (p.isAuthenticated) {
+        // TODO: use name and email from backend
+        String name = p.googleSignIn.currentUser.displayName;
+        String email = p.googleSignIn.currentUser.email;
+        String imageUrl = p.googleSignIn.currentUser.photoUrl;
+        String driverID = p.id;
+        return Home(name, email, imageUrl, driverID);
```

Comment:
The whole point of the AuthProvider is to avoid the passing of arguments to other widgets. Change Home to use AuthProvider provided data instead of being dependent on these arguments.

Reply:
Make the necessary changes to other widgets that take arguments as well. If the profile page, for example, requires arguments, remove them and use the AuthProvider instead. Do this everywhere that is possible. Remove any import that become irrelevant due to this refactoring process. The app should have no global variable anywhere. Check for that as well.

Reply:
Edit: Definitely remove unnecessary imports as then your branch cannot be merged as it will not pass the build check of GitHub.